lib/rspec/matchers/expecteds_for_multiple_diffs.rb

```diff
@@ -46,7 +46,12 @@ def self.for_many_matchers(matchers)
       # @return [String]
       def message_with_diff(message, differ, actual)
         diff = diffs(differ, actual)
-        message = "#{message}\n#{diff}" unless diff.empty?
+        if @expected_list[0].first.class != actual.class
```

Comment:
Diff is only printed if the expected is not equal to the actual, hence my earlier suggestion of simply printing a warning message when the diff is empty, we don't need to detect anything else.

Reply:
That's a good suggestion. I will simplify my PR. Thanks Jon